File: run_supabase_fund_scraper.py
#!/usr/bin/env python3
"""
Python wrapper for running the Supabase Fund Scraper
Executes the Node.js scraper with proper configuration
"""

import subprocess
import sys
import os
import json
from datetime import datetime

def run_fund_scraper(max_pages=None, start_page=1, enable_detailed_scraping=True):
    """Run the Supabase fund scraper with specified parameters"""
    
    print("🏦 DIVA Fund Scraper for Supabase")
    print("=" * 60)
    print(f"📍 Enhanced with 대표펀드매니저 (Fund Manager) Data Extraction")
    print(f"📅 Started: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    print()
    
    # Configuration
    config = {
        'maxPages': max_pages,
        'startPage': start_page,
        'enableDetailedScraping': enable_detailed_scraping,
        'batchSize': 20,
        'delayBetweenRequests': 100,
        'delayBetweenBatches': 300
    }
    
    print(f"⚙️ Configuration:")
    print(f"   • Max Pages: {max_pages or 'ALL'}")
    print(f"   • Start Page: {start_page}")
    print(f"   • Enhanced Scraping: {enable_detailed_scraping}")
    print(f"   • Target: Supabase fund_table")
    print()
    
    # Check if Node.js is available
    try:
        subprocess.run(['node', '--version'], check=True, capture_output=True)
    except (subprocess.CalledProcessError, FileNotFoundError):
        print("❌ Node.js is required but not found. Please install Node.js first.")
        return False
    
    # Check if the scraper file exists
    scraper_path = 'src/supabase/supabase_fund_scraper.js'
    if not os.path.exists(scraper_path):
        print(f"❌ Scraper file not found: {scraper_path}")
        return False
    
    try:
        print("🚀 Starting fund scraper...")
        
        # Set environment variables for configuration
        env = os.environ.copy()
        env['SCRAPER_CONFIG'] = json.dumps(config)
        
        # Run the Node.js scraper
        process = subprocess.run(
            ['node', scraper_path],
            env=env,
            capture_output=False,  # Show real-time output
            text=True
        )
        
        if process.returncode == 0:
            print("\n✅ Fund scraper completed successfully!")
            print("📊 Check Supabase fund_table for new records with enhanced manager data")
            return True
        else:
            print(f"\n❌ Fund scraper failed with exit code: {process.returncode}")
            return False
            
    except Exception as e:
        print(f"❌ Error running fund scraper: {e}")
        return False

def main():
    """Main execution function"""
    import argparse
    
    parser = argparse.ArgumentParser(description='Run Supabase Fund Scraper')
    parser.add_argument('--max-pages', type=int, help='Maximum pages to scrape (default: all)')
    parser.add_argument('--start-page', type=int, default=1, help='Starting page number')
    parser.add_argument('--no-detailed-scraping', action='store_true', help='Disable detailed fund manager scraping')
    
    args = parser.parse_args()
    
    # Run the scraper
    success = run_fund_scraper(
        max_pages=args.max_pages,
        start_page=args.start_page,
        enable_detailed_scraping=not args.no_detailed_scraping
    )
    
    if success:
        print("\n🎉 Fund data successfully scraped to Supabase!")
        print("💡 You can now use comprehensive_vc_openai_interface.py to query fund manager information")
    else:
        print("\n❌ Fund scraping failed. Check the error messages above.")
        sys.exit(1)

if __name__ == "__main__":
    main() 

File: run_supabase_fund_scraper.js
/**
 * 🏦 APIFY FUND SCRAPER RUNNER - SUPABASE VERSION
 * ===============================================
 * 
 * Runs the Fund scraper via Apify API and sends data to Supabase
 * Based on your working run_scraper_api_funds_airtable.js pattern
 */

const { ApifyClient } = require('apify-client');
const fs = require('fs').promises;

// Load environment variables
require('dotenv').config();

// Get environment variables
const APIFY_API_TOKEN = process.env.APIFY_API_TOKEN;
const SUPABASE_URL = process.env.SUPABASE_URL;
const SUPABASE_KEY = process.env.SUPABASE_KEY;

if (!APIFY_API_TOKEN) {
    console.error('❌ Error: APIFY_API_TOKEN environment variable is not set');
    console.error('Please set it using: $env:APIFY_API_TOKEN="your-token-here"');
    process.exit(1);
}

if (!SUPABASE_URL || !SUPABASE_KEY) {
    console.error('❌ Error: SUPABASE_URL and SUPABASE_KEY environment variables must be set');
    console.error('Please set them in your .env file');
    process.exit(1);
}

// Initialize the ApifyClient with API token
const client = new ApifyClient({
    token: APIFY_API_TOKEN,
});

async function runFundScraper() {
    try {
        console.log('🚀 Starting Fund scraper for Supabase...');

        // Read the working Supabase Fund scraper code
        const scraperCode = await fs.readFile('supabase_fund_scraper_working.js', 'utf8');

        // Run your custom scraper actor (Fund data for Supabase)
        const run = await client.actor("ECFcJkFqhtxYGqqGy").call({
            code: scraperCode,
            dataType: 'Fund',
            maxPages: 5, // Start with 5 pages for testing
            
            // Pass Supabase credentials as input
            supabaseUrl: SUPABASE_URL,
            supabaseKey: SUPABASE_KEY
        });

        console.log('✅ Fund scraper finished successfully!');
        console.log('📊 Run ID:', run.id);
        console.log('📋 Check Apify Console for detailed logs');
        console.log('🗄️ Check Supabase fund_table for imported data');

        // Get basic stats
        if (run.defaultDatasetId) {
            try {
                const { items } = await client.dataset(run.defaultDatasetId).listItems();
                console.log(`📈 Dataset contains ${items.length} records`);
                
                if (items.length > 0) {
                    console.log('🔍 Sample record structure:');
                    console.log(JSON.stringify(items[0], null, 2));
                }
            } catch (datasetError) {
                console.log('⚠️ Could not fetch dataset info:', datasetError.message);
            }
        }

    } catch (error) {
        console.error('💥 Error running Fund scraper:', error.message);
        process.exit(1);
    }
}

// Execute the scraper
runFundScraper(); 